src/foam/u2/crunch/CrunchController.js

```diff
@@ -243,6 +264,51 @@ foam.CLASS({
           functionData: capabilitiesSections,
           capabilityId: capabilityId
         })).end();
+    },
+    function save(wizardlet) {
+      var self = wizardlet;
+      var isAssociationCapability = foam.nanos.crunch.AssociationCapability.isInstance(self.capability);
+      var associatedEntity = isAssociationCapability ? self.subject.realUser : 
+        self.capability.associatedEntity === 'user' ? self.subject.user : self.subject.realUser;
+
+      return this.updateUCJ(self, associatedEntity).then(() => {
+        var ucj = self.ucj;
+        if ( ucj === null ) {
+          ucj = isAssociationCapability ? 
+          self.AgentCapabilityJunction.create({
+              sourceId: associatedEntity.id,
+              targetId: self.capability.id,
+              effectiveUser: self.subject.user.id
+            })
+            : self.UserCapabilityJunction.create({
+              sourceId: associatedEntity.id,
+              targetId: self.capability.id
+            })
+        }
+        if ( self.of ) ucj.data = self.data;
+        return self.userCapabilityJunctionDAO.put(ucj);
+      });
+    }, 
+    async function updateUCJ(self, associatedEntity) {
+      return self.userCapabilityJunctionDAO.find(
+        self.AND(
+          self.OR(
+            self.AND(
+              self.NOT(self.INSTANCE_OF(self.AgentCapabilityJunction)),
+              self.EQ(self.UserCapabilityJunction.SOURCE_ID, associatedEntity.id)
+            ),
+            self.AND(
+              self.INSTANCE_OF(self.AgentCapabilityJunction),
+              self.EQ(self.UserCapabilityJunction.SOURCE_ID, associatedEntity.id),
+              self.EQ(self.AgentCapabilityJunction.EFFECTIVE_USER, self.subject.user.id)
+            )
+          ),
+          self.EQ(self.UserCapabilityJunction.TARGET_ID, self.capability.id)
+        )
+      ).then(ucj => {
+        self.ucj = ucj;
```

Comment:
hmmm - self.ucj ? 

CrunchController should not be saving any state thus should not be storing a ucj.

Reply:
self refers to the wizardlet passed in line 269, working on a commit to use imports and requires from crunchcontroller instead of those from capabilitywizardlet and rename self back to wizardlet

Reply:
resolved in https://github.com/foam-framework/foam2/pull/3672/commits/df543b6c18e30c6cf5e56c0a5b918af15ca5ef13